rs/ledger_suite/icrc1/ledger/src/main.rs

```diff
@@ -146,6 +156,16 @@ fn pre_upgrade() {
     });
 }
 
+#[cfg(not(feature = "low-upgrade-instruction-limits"))]
+const MAX_INSTRUCTIONS_PER_UPGRADE: u64 = 199_950_000_000;
+#[cfg(not(feature = "low-upgrade-instruction-limits"))]
+const MAX_INSTRUCTIONS_PER_TIMER_CALL: u64 = 1_950_000_000;
+
+#[cfg(feature = "low-upgrade-instruction-limits")]
+const MAX_INSTRUCTIONS_PER_UPGRADE: u64 = 0;
```

Comment:
Shouldn't we have a positive value here so that migration is partially performed in the post-upgrade hook in tests?

Reply:
The problem I ran into is that this number would have to cover the upgrade + some migration which would be hard to set right and could be flaky if we depend on some elements being migrated in post_upgrade. Also, there would currently be no way of telling if some elements were migrated in post_upgrade - I only count the number of steps but not if the steps migrated something or exited immediately due to instruction limit.

Reply:
I see your point, but this deviation from mainnet limits causes a qualitative change in behavior which is not tested. Should we then maybe never perform any migration in the post-upgrade hook?

Reply:
> Do you think it makes sense to add an additional metric to be able to add an assert?

It'd make sense in my opinion if it doesn't add too much extra work: could you query the state right after post_upgrade and before the first timer to assert without adding a new metric?

Reply:
I tried that, but even querying directly after upgrade returns more than 1 step.

Reply:
I see, I suppose the timer fires in the same round as the upgrade then. A manual test is enough then.